modules/javafx.controls/src/test/java/test/javafx/scene/control/LabeledTruncatedTest.java

```diff
@@ -0,0 +1,95 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.javafx.scene.control;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import javafx.scene.control.Label;
+import javafx.scene.control.Labeled;
+import javafx.scene.control.TableCell;
+import javafx.scene.control.TreeTableCell;
+import javafx.scene.control.skin.LabelSkin;
+import javafx.scene.control.skin.TableCellSkin;
+import javafx.scene.control.skin.TreeTableCellSkin;
+import org.junit.jupiter.api.Test;
+import com.sun.javafx.scene.layout.RegionHelper;
+import com.sun.javafx.tk.Toolkit;
+
+/**
+ * Tests textTruncated property of Labeled, using Label, TableCell, and TreeTableCell controls
+ * (the last two contain conditional code that redirects the execution of computePrefWidth())
+ * in their skins to different code paths.
+ */
+public class LabeledTruncatedTest {
+    private static final String TEXT = "testing.truncated";
+
+    private void firePulse() {
+        Toolkit.getToolkit().firePulse();
+    }
+
+    @Test
+    public void testTruncatedLabel2() {
+        Label control = new Label();
+        control.setSkin(new LabelSkin(control));
+        control.setText(TEXT);
+        test(control);
+    }
+
+    @Test
+    public void testTruncatedTableCellSkin() {
+        TableCell<String, String> control = new TableCell<>();
+        control.setSkin(new TableCellSkin<>(control));
+        control.setText(TEXT);
+        test(control);
+    }
+
+    @Test
+    public void testTruncatedTreeTableCellSkin() {
+        TreeTableCell<String, String> control = new TreeTableCell<>();
+        control.setSkin(new TreeTableCellSkin<>(control));
+        control.setText(TEXT);
+        test(control);
+    }
+
+    private void test(Labeled control) {
+        RegionHelper.setWidth(control, 1000);
+        firePulse();
+        double w = control.prefWidth(-1);
+        double h = control.prefHeight(-1);
+
+        assertFalse(control.isTextTruncated());
+
+        RegionHelper.setWidth(control, 10);
+        RegionHelper.setHeight(control, h);
+        firePulse();
+
+        assertTrue(control.isTextTruncated());
+
+        control.setWrapText(true);
+        firePulse();
+
+        assertFalse(control.isTextTruncated());
```

Comment:
Can you add a test for the case where we are wrapping and the preferred height is exceeded?